index.ts

```diff
@@ -138,17 +138,19 @@ function $$<Selected extends Element>(
 	// Can be: select.all('selectors') or select.all('selectors', singleElementOrDocument)
 	if (!baseElements || isQueryable(baseElements)) {
 		const elements = (baseElements ?? document).querySelectorAll<Selected>(String(selectors));
-		return Array.prototype.slice.call(elements) as Selected[];
+		return Array.from(elements) as Selected[];
 	}
 
-	const elements = new Set<Selected>();
+	const elements = Selected[];
 	for (const baseElement of baseElements) {
 		for (const element of baseElement.querySelectorAll<Selected>(String(selectors))) {
-			elements.add(element);
+			if (!elements.includes(element)) {
+				elements.push(element);
+			}
```

Comment:
.includes() has a cost too. In my browser set.add is more than twice as fast as .includes+push: https://measurethat.net/Benchmarks/Show/17222/0/setadd-vs-arraypush-vs-mapset

To claim that this actually speeds up the code you’d have to back it up with performance tests like that link (I just found it online)